axonn/communication.py

```diff
@@ -31,6 +31,17 @@ def __init__(
         """
         self.world_rank = MPI.COMM_WORLD.Get_rank()
         self.world_size = MPI.COMM_WORLD.Get_size()
+        
+        if gpus_per_node is None:
+            self.backend = 'gloo'
+            self.is_gpu_available = False
+            self.local_rank = os.environ.get("SLURM_LOCALID", 0)
```

Comment:
I don't think `local_rank` gets used if it's on cpu, so we can probably get rid of this.